src/cascadia/TerminalApp/AppLogic.idl

```diff
@@ -50,94 +27,22 @@ namespace TerminalApp
         Boolean HasSettingsStartupActions();
-        Int32 SetStartupCommandline(String[] commands);
-        Int32 ExecuteCommandline(String[] commands, String cwd);
-        String ParseCommandlineMessage { get; };
-        Boolean ShouldExitEarly { get; };
-
-        void Quit();
-
-        void ReloadSettings();
-        Windows.UI.Xaml.UIElement GetRoot();
-
-        void SetInboundListener();
-
-        String Title { get; };
-
-        Boolean FocusMode { get; };
-        Boolean Fullscreen { get; };
-        void Maximized(Boolean newMaximized); 
-        Boolean AlwaysOnTop { get; };
-        Boolean AutoHideWindow { get; };
-
-        void IdentifyWindow();
-        String WindowName;
-        UInt64 WindowId;
-        void SetPersistedLayoutIdx(UInt32 idx);
-        void SetNumberOfOpenWindows(UInt64 num);
-        void RenameFailed();
-        void RequestExitFullscreen();
-        Boolean IsQuakeWindow();
-
-        Windows.Foundation.Size GetLaunchDimensions(UInt32 dpi);
-        Boolean CenterOnLaunch { get; };
-
-        InitialPosition GetInitialPosition(Int64 defaultInitialX, Int64 defaultInitialY);
-        Windows.UI.Xaml.ElementTheme GetRequestedTheme();
-        Microsoft.Terminal.Settings.Model.LaunchMode GetLaunchMode();
-        Boolean GetShowTabsInTitlebar();
-        Boolean GetInitialAlwaysOnTop();
-        Single CalcSnappedDimension(Boolean widthOrHeight, Single dimension);
-        void TitlebarClicked();
-        void CloseWindow(Microsoft.Terminal.Settings.Model.LaunchPosition position);
-        void WindowVisibilityChanged(Boolean showOrHide);
-
-        TaskbarState TaskbarState{ get; };
-        Windows.UI.Xaml.Media.Brush TitlebarBrush { get; };
-        void WindowActivated(Boolean activated);
 
         Boolean ShouldUsePersistedLayout();
-        Boolean ShouldImmediatelyHandoffToElevated();
-        void HandoffToElevated();
-        String GetWindowLayoutJson(Microsoft.Terminal.Settings.Model.LaunchPosition position);
         void SaveWindowLayoutJsons(Windows.Foundation.Collections.IVector<String> layouts);
 
-        Boolean GetMinimizeToNotificationArea();
-        Boolean GetAlwaysShowNotificationIcon();
-        Boolean GetShowTitleInTitlebar();
+        void ReloadSettings();
 
         Microsoft.Terminal.Settings.Model.Theme Theme { get; };
 
         FindTargetWindowResult FindTargetWindow(String[] args);
 
+        TerminalWindow CreateNewWindow();
+
         Windows.Foundation.Collections.IMapView<Microsoft.Terminal.Control.KeyChord, Microsoft.Terminal.Settings.Model.Command> GlobalHotkeys();
 
-        // See IDialogPresenter and TerminalPage's DialogPresenter for more
-        // information.
-        Windows.Foundation.IAsyncOperation<Windows.UI.Xaml.Controls.ContentDialogResult> ShowDialog(Windows.UI.Xaml.Controls.ContentDialog dialog);
-        void DismissDialog();
+        event Windows.Foundation.TypedEventHandler<Object, SettingsLoadEventArgs> SettingsChanged;

```

Comment:
```mermaid
sequenceDiagram
    participant Emperor
    participant AppLogic
    
    participant AppHost
    participant TerminalWindow
    participant TerminalPage

    Note Right of AppLogic: AL::ReloadSettings
    AppLogic ->> Emperor: raise SettingsChanged
    Note left of Emperor: E::...GlobalHotkeys
    Note left of Emperor: E::...NotificationIcon
    AppLogic ->> TerminalWindow: raise SettingsChanged<br>(to each window)
    AppLogic ->> TerminalWindow: 
    AppLogic ->> TerminalWindow: 
    Note right of TerminalWindow: TW::UpdateSettingsHandler
    Note right of TerminalWindow: TW::UpdateSettings
    TerminalWindow ->> TerminalPage: SetSettings
    TerminalWindow ->> AppHost: raise SettingsChanged
    Note right of AppHost: AH::_HandleSettingsChanged

    
```

Reply:
Awesome. And out of curiosity, does this mean there's only one copy of the settings model around that everybody queries? Or do we have one for each `TerminalWindow`?

Reply:
Ah, here's a relevant excerpt from the next PR:
> The emperor can do that - there's only ever one emperor. It can also own a singular copy of the settings model, and hand out references to each other thread.

so yeah, guess that answers that. Nice!